modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/CacheTableDescriptorImpl.java

```diff
@@ -309,7 +308,11 @@ else if (affFields.isEmpty())
 
         Object dfltVal = TypeUtils.toInternal(dataCtx, desc.defaultValue());
 
-        return rexBuilder.makeLiteral(dfltVal, desc.logicalType(typeFactory), false);
+        if (dfltVal instanceof UUID)
+            return rexBuilder.makeCast(typeFactory.createUUIDType(),
+                rexBuilder.makeLiteral(dfltVal.toString(), desc.logicalType(typeFactory), false));
+
+        return rexBuilder.makeLiteral(dfltVal, desc.logicalType(typeFactory), true);
```

Comment:
Let's move it to the new method of `TypeUtils` class (for example, `toRexLiteral`)

Reply:
Done. But not sure. Many arguments to pass into the new method.